Content.Server/StationEvents/Events/CargoGiftsRule.cs

```diff
@@ -0,0 +1,79 @@
+﻿using System.Linq;
+using Content.Server.Anomaly;
+using Content.Server.Cargo.Components;
+using Content.Server.Cargo.Systems;
+using Content.Server.GameTicking.Rules.Components;
+using Content.Server.Station.Components;
+using Content.Server.Station.Systems;
+using Content.Server.StationEvents.Components;
+using Content.Shared.Access.Components;
+using Content.Shared.Administration.Logs;
+using Content.Shared.Cargo;
+using Content.Shared.Cargo.Prototypes;
+using Content.Shared.Database;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Random;
+using Robust.Shared.Utility;
+
+namespace Content.Server.StationEvents.Events;
+
+public sealed class CargoGiftsRule : StationEventSystem<CargoGiftsRuleComponent>
+{
+    [Dependency] private readonly CargoSystem _cargoSystem = default!;
+    [Dependency] private readonly StationSystem _stationSystem = default!;
+    [Dependency] private readonly ISharedAdminLogManager _adminLogger = default!;
+    [Dependency] private readonly IPrototypeManager _prototypeManager = default!;
+
+    protected override void Added(EntityUid uid, CargoGiftsRuleComponent component, GameRuleComponent gameRule, GameRuleAddedEvent args)
+    {
+        base.Added(uid, component, gameRule, args);
+
+        var str = Loc.GetString("cargo-gifts-event-announcement",
+            ("sender", Loc.GetString(component.Sender)), ("description", Loc.GetString(component.Description)), ("dest", Loc.GetString(component.Dest)));
+        ChatSystem.DispatchGlobalAnnouncement(str, colorOverride: Color.FromHex("#18abf5"));
+    }
+
+    /// <summary>
+    /// Called on an active gamerule entity in the Update function
+    /// </summary>
+    protected override void ActiveTick(EntityUid uid, CargoGiftsRuleComponent component, GameRuleComponent gameRule, float frameTime)
+    {
+        if (component.Gifts.Count == 0)
+        {
+            return;
+        }
+
+        if (component.TimeUntilNextGifts > 0)
+        {
+            component.TimeUntilNextGifts -= frameTime;
+            return;
+        }
+        component.TimeUntilNextGifts = 30f;
+
+        var station = _stationSystem.Stations.FirstOrNull();
+        if (station == null)
+        {
+            return;
+        }
+
+        if (!TryComp<StationCargoOrderDatabaseComponent>(station, out var cargoDb))
+        {
+            return;
+        }
+
+        // Add some presents
+        int outstanding = _cargoSystem.GetOutstandingOrderCount(cargoDb);
+        while (outstanding < cargoDb.Capacity - component.OrderSpaceToLeave && component.Gifts.Count > 0)
+        {
+            // I wish there was a nice way to pop this
+            var (productId, qty) = component.Gifts.First();
+            component.Gifts.Remove(productId);
```

Comment:
Aren't gamerules singletons? It means the event can never run again which seems suss.

Reply:
They are spawned entities, generally created and destroyed by the random game system.

Reply:
I ran the party once, received the goods via shuttle, then ran it again and it works OK.

![image](https://github.com/space-wizards/space-station-14/assets/5285589/79fc2ad1-6c7c-45d4-b6c9-8c009ac9ca33)